src/mid_market_rate.rs

```diff
@@ -64,8 +64,10 @@ mod kraken {
         type Error = anyhow::Error;
 
         fn try_from(AskAndBid { ask, bid }: AskAndBid) -> anyhow::Result<Self> {
-            // TODO: Fix the ceil to precision of 9
-            let value = ((bid + ask) / 2f64).ceil();
+            let value = (bid + ask) / 2f64;
```

Comment:
Should we try to avoid this floating arithmetics?

Reply:
What do you propose instead? And what's the harm in using floating point arithmetic here? I'm a computer maths noob.

Reply:
The result might not be the one to expect if it cannot be stored in 64 (binary) bytes

Reply:
But wouldn't it just be a matter of losing a little bit of precision? In any case, I'm not too sure how I would go about avoiding this...

Reply:
> In any case, I'm not too sure how I would go about avoiding this...

I could try to use a dependency such as https://docs.rs/fixed/1.1.0/fixed/ if you're okay with it.

Reply:
I am fine with introducing a fixed-point library, I think in the long run it would probably make our life easier. 
Might be a good point to to try the library!

Note: I remember that there was quite some discussion around "use a library and trust it" vs. "write it ourselves to trust it" - alternatively we can also introduce another function for handling this division into `float_maths` (similar to `divide_pow_ten_trunc`).

Reply:
Okay, thanks for the feedback, I'll handle it in a separate PR :wink: